gym/spaces/box.py

```diff
@@ -56,7 +56,7 @@ def __init__(
         low: Union[SupportsFloat, np.ndarray],
         high: Union[SupportsFloat, np.ndarray],
         shape: Optional[Sequence[int]] = None,
-        dtype: Type = np.float32,
+        dtype: Union[Type, object] = np.float32,
```

Comment:
I'm not so sure about this annotation, since pretty much anything is an `object` anyways (maaaaaaybe except for some extremely exotic metaclass constructs, I'm not sure)

We want it to be either a python type, or a numpy datatype, right? I think numpy has a type annotation for this somewhere in its typing module

Reply:
The problem is that `np.float32` is not actually a type but an object and `np.result_type()` (in utils.py L322) returns an object

Reply:
Changed back to `Type`